### DISCUSSION TITLE
Prometheus metrics reset on cluster restart

### DISCUSSION THREAD

**Discussion by broferek** [minio/minio#16644] [Q&A]
Hi,

Following a discussion on Slack, I was advised to open a discussion here.

So, my problem is that both minio_bucket_usage_object_total and minio_bucket_usage_total_bytes Prometheus metrics are reset on deployment restart.

As I often update the cluster with the latest release, I end up having the following graph in Prometheus (for instance for the per bucket objects evolution) :
![image](https://user-images.githubusercontent.com/2465124/219729923-04bebd02-0ce4-4023-b2e5-ad41c58209a1.png)

As you can see, it is hard to know the real number of objects in my bucket over time and the lastest metric doesn't reflect the reality at all.

My guess is that these metrics should survive a deployment restart in order to prevent these drops in the graphs.

Any feedback on this would really be much appreciated.

Looking forward to hearing from you.

Thanks,

Ludovic

**Comment by harshavardhana** [comment]
They are of course preserved, which version of MinIO are you using ?

**Reply by broferek** [reply]
I'm on a very recent one : RELEASE.2023-01-31T02-24-19Z

**Reply by harshavardhana** [reply]
This is not recent however from which version to which version did you upgrade?

**Reply by broferek** [reply]
The first significant drop you see on the graph (around 02/04) occured when upgrading from RELEASE.2023-01-25T00-19-54Z to RELEASE.2023-01-31T02-24-19Z.

The second drop around 02/16 occured when rebooting the servers. No upgrade was carried out at that time.

Another important point to notice is that I also made an upgrade from RELEASE.2023-01-20T02-05-44Z to RELEASE.2023-01-25T00-19-54Z on Jan 25 and by that time, the metrics were preserved.

**Reply by harshavardhana** [reply]
Did you have any disk failures? how many disks do you have?

**Reply by broferek** [reply]
Indeed, we had a healing disk from Jan 24 to Feb 11.

We have a pool of 8 servers running with 2 drives of 6TB each.

**Reply by harshavardhana** [reply]
I don't know but this shouldn't really be possible unless the usage scanner saw some incorrect results with the persisted file.

**Reply by broferek** [reply]
Do you see anything I could do to force it to calculate the total number of objects and their size again ?

**Reply by harshavardhana** [reply]
You should upgrade there was a bug that was fixed in the last release that would cause metrics to be blocked.

**Reply by broferek** [reply]
OK, I'll do that in the coming week and let you know if it's fixed.

**Comment by wokalski** [comment]
I have upgraded to the latest version and I'm still seeing the same. "Object size distribution" is showing the correct values (minio_bucket_objects_size_distribution) but `minio_bucket_usage_total_bytes` is incorrect. When I went to the MinIO console though, the object counts are incorrect for one of the buckets.

I am on [RELEASE.2023-06-19T19-52-50Z](https://github.com/minio/minio/releases/tag/RELEASE.2023-06-19T19-52-50Z) but the metrics got corrupted yesterday while I was using an older version and my cluster was misbehaving (due to hardware failures).

**Reply by klauspost** [reply]
This will self-correct.

**Reply by wokalski** [reply]
it is gradually self correcting

**Reply by klauspost** [reply]
Yes. All these stats are lazy. If some are corrupted, they will be corrected on the next scan.
